Source/Core/joinUrls.js

```diff
@@ -55,6 +57,9 @@ define([
         var baseUri = first;
         if (second.isAbsolute()) {
             baseUri = second;
+            if (second.scheme === 'data') {
```

Comment:
What is this for?  It's probably worth it to have a comment here, since the purpose of this isn't immediately obvious.